Doc/c-api/unicode.rst

```diff
@@ -1396,10 +1396,12 @@ They all return ``NULL`` or ``-1`` if an exception occurs.
    :c:func:`PyErr_Occurred` to check for errors.
 
 
-.. c:function:: int PyUnicode_EqualToString(PyObject *unicode, const char *string)
+.. c:function:: int PyUnicode_EqualToUTF8(PyObject *unicode, const char *string)
 
    Compare a Unicode object with a UTF-8 encoded C string and return true
    if they are equal and false otherwise.
+   If the Unicode object contains null or surrogate characters or
+   the C string not encoded to UTF-8 return false.
 
    This function does not raise exceptions.
```

Comment:
Can you say something to repeat that passing ASCII strings are perfectly fine?

Reply:
Do we have something similar in other functions that take `const char*` argument?

Reply:
I don't think that documentation of other functions say something about "C string is not encoded to UTF-8". Hum. I propose to replace:

```
the C string is not encoded to UTF-8 return false.
```

with:

```
the C string is not encoded to UTF-8 nor to ASCII return false.
```

I'm not sure that all users are aware that UTF-8 is a superset of ASCII.

Reply:
Should not we add "or ASCII" to every "UTF-8" in the C API documentation.